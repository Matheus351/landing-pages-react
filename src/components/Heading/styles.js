import styled, { css } from 'styled-components';
// import {theme} from '../../styles/theme';

const titleSize = {
  small: (theme) =>
    css`
      font-size: ${theme.sizes.small};
    `,
  medium: (theme) =>
    css`
      font-size: ${theme.sizes.medium};
    `,
  big: (theme) =>
    css`
      font-size: ${theme.sizes.xlarge};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.sizes.xhuge};
      ${mediaFont(theme)};
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.white : theme.colors.primaryColor};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
  `}
`;
