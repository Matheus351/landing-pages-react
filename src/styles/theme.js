export const theme = {
  colors: {
    mainBg: 'white',
    secondaryBg: 'grey',
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
  },
  fonts: {
    default: "'Open Sans', sans-serif;",
    secondary: "'Montserrat', sans-serif;",
  },
  sizes: {
    xsmal: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.2rem',
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
};
