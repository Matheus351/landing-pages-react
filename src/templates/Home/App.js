function Home() {
  return (
    <div className="App">
      <h2>Oi</h2>
    </div>
  );
}

export default Home;
