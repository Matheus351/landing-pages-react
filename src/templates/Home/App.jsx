import * as Styled from './styles';

function Home() {
  return (
    <div className="App">
      <Styled.Wrapper>
        <h2>Oi</h2>
      </Styled.Wrapper>
    </div>
  );
}
export default Home;
