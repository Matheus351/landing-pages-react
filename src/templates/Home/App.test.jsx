import { render, screen } from '@testing-library/react';
import Home from './App';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';

test('renders learn react link', () => {
  const { debug } = renderTheme(<Home />);
  const headingContainer = screen.getByRole('heading', {
    name: 'Oi',
  }).parentElement;
  expect(headingContainer).toHaveStyle({
    background: theme.colors.mainBg,
  });
  expect(headingContainer).toMatchSnapshot();
  expect(headingContainer).toHaveStyleRule('background', 'cyan');
  debug();
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
