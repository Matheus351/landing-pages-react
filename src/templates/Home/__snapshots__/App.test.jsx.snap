// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c0 {
  background: cyan;
}

<div
  class="c0"
>
  <h2>
    Oi
  </h2>
</div>
`;
